src/services/firebase/index.ts

```diff
@@ -0,0 +1,257 @@
+import { formatUnits } from 'ethers/lib/utils'
+import { get } from 'idb-keyval'
+import type { MessagePayload } from 'firebase/messaging/sw'
+import type { ChainInfo, SafeBalanceResponse, ChainListResponse } from '@safe-global/safe-gateway-typescript-sdk'
+
+import { shortenAddress } from '@/utils/formatters'
+import { AppRoutes } from '@/config/routes'
+import { isWebhookEvent, WebhookType } from '@/services/firebase/webhooks'
+import type { WebhookEvent } from '@/services/firebase/webhooks'
+import { GATEWAY_URL_PRODUCTION, GATEWAY_URL_STAGING, IS_PRODUCTION } from '@/config/constants'
+import {
+  createPreferencesStore,
+  getSafeNotificationKey,
+} from '@/components/settings/PushNotifications/hooks/notifications-idb'
+import type {
+  NotificationPreferences,
+  SafeNotificationKey,
+} from '@/components/settings/PushNotifications/hooks/notifications-idb'
+
+export const shouldShowNotification = async (payload: MessagePayload): Promise<boolean> => {
+  if (!isWebhookEvent(payload.data)) {
+    return true
+  }
+
+  const { chainId, address, type } = payload.data
+
+  const key = getSafeNotificationKey(chainId, address)
+  const store = createPreferencesStore()
+
+  const preferencesStore = await get<NotificationPreferences[SafeNotificationKey]>(key, store).catch(() => null)
+
+  if (!preferencesStore) {
+    return false
+  }
+
+  return preferencesStore.preferences[type]
+}
+
+// localStorage cannot be accessed in service workers so we reference the flag
+const BASE_URL = IS_PRODUCTION ? GATEWAY_URL_PRODUCTION : GATEWAY_URL_STAGING
+
+// XHR is not supported in service workers so we can't use the SDK
+const getChains = async (): Promise<ChainListResponse | undefined> => {
+  const ENDPOINT = `${BASE_URL}/v1/chains`
+
+  const response = await fetch(ENDPOINT)
+
+  if (response.ok) {
+    return response.json()
+  }
+}
+
+const getBalances = async (chainId: string, safeAddress: string): Promise<SafeBalanceResponse | undefined> => {
+  const DEFAULT_CURRENCY = 'USD'
+  const ENDPOINT = `${BASE_URL}/v1/chains/${chainId}/safes/${safeAddress}/balances/${DEFAULT_CURRENCY}`
+
+  const response = await fetch(ENDPOINT)
+
+  if (response.ok) {
+    return response.json()
+  }
+}
+
+const getLink = (path: string, query: { address: string; chain?: ChainInfo; safeTxHash?: string }) => {
+  const APP_URL = 'https://app.safe.global'
```

Comment:
Can we somehow get the actual running app url? 
So localhost in dev / the PR deployments URLs?